Data/Vector.hs

```diff
@@ -1530,38 +1530,86 @@ product = G.product
 
 -- | /O(n)/ Yield the maximum element of the vector. The vector may not be
 -- empty.
+--
+-- ==== __Examples__
+--
+-- >>> import qualified Data.Vector as V
+-- >>> V.maximum $ V.fromList [2.0, 1.0]
```

Comment:
Minor improvement to show that `maximum` for generic vectors works with other type of vectors:
```suggestion
-- >>> import qualified Data.Vector.Generic as VG
-- >>> import qualified Data.Vector as V
-- >>> VG.maximum $ V.fromList ([2.0, 1.0] :: [Integer])
-- >>> import qualified Data.Vector as VS
-- >>> VG.maximum $ VS.fromList ([1.0, 2.0, 3.0] :: [Int])
```

Reply:
I think it would be better to explain this in top-level documentation of modules. Otherwise some example will show this difference and some won't.